boilerplate_app/views.py

```diff
@@ -185,4 +186,40 @@ def post(self, request, format=None):
                                 status=status.HTTP_400_BAD_REQUEST)
         except Exception as e:
             return Response({'status': False, 'message': str(e)},
-                            status=status.HTTP_400_BAD_REQUEST)
\ No newline at end of file
+                            status=status.HTTP_400_BAD_REQUEST)
+        
+
+class PunchInAPIView(CreateAPIView):
+    serializer_class = TimeEntryCreateSerializer
+    #permission_classes = (IsAuthenticated,)
+
+    def post(self, request, format=None):
+        """
+        punch in
+        """
+        
+        try:
+            data = request.data
+            #data['account'] = request.account.pk
+            serializer = TimeEntryCreateSerializer(data = data)
+
+            if serializer.is_valid():
+                timeEntry = serializer.create(serializer.data)
```

Comment:
quick question. whose `timeEntry` is this and how does the database know that?

Reply:
ok, you handled this in the serializer. cool!

Reply:
but my preference is to have everything handled in the apiview. serializer is just used to serialize raw data from the requests.